crates/turborepo-lib/src/task_graph/visitor/command.rs

```diff
@@ -155,37 +159,29 @@ impl<'a> CommandProvider for PackageGraphCommandProvider<'a> {
 }
 
 #[derive(Debug)]
-pub struct MicroFrontendProxyProvider<'a> {
+pub struct MicroFrontendProxyProvider<'a, T> {
     repo_root: &'a AbsoluteSystemPath,
-    package_graph: &'a PackageGraph,
+    package_graph: &'a T,
     tasks_in_graph: HashSet<TaskId<'a>>,
     mfe_configs: &'a MicrofrontendsConfigs,
 }
 
-impl<'a> MicroFrontendProxyProvider<'a> {
-    pub fn new(
+impl<'a, T: PackageInfoProvider> MicroFrontendProxyProvider<'a, T> {
+    pub fn new<'b>(
         repo_root: &'a AbsoluteSystemPath,
-        package_graph: &'a PackageGraph,
-        engine: &Engine,
+        package_graph: &'a T,
+        tasks_in_graph: impl Iterator<Item = &'b TaskId<'static>>,
         micro_frontends_configs: &'a MicrofrontendsConfigs,
     ) -> Self {
-        let tasks_in_graph = engine
-            .tasks()
-            .filter_map(|task| match task {
-                crate::engine::TaskNode::Task(task_id) => Some(task_id),
-                crate::engine::TaskNode::Root => None,
-            })
-            .cloned()
-            .collect();
         Self {
             repo_root,
             package_graph,
-            tasks_in_graph,
+            tasks_in_graph: tasks_in_graph.cloned().collect(),
```

Comment:
Moved this logic over to `Engine` so we could test this command provider without building an entire task graph.

Reply:
Nice